djmoney/contrib/exchange/models.py

```diff
@@ -64,17 +55,20 @@ def get_rate(source, target, backend=None):
 def _get_rate(source, target, backend):
     if text_type(source) == text_type(target):
         return 1
-    try:
-        forward = models.Q(currency=target, backend__base_currency=source)
-        reverse = models.Q(currency=source, backend__base_currency=target)
-        return Rate.objects.annotate(
-            rate=models.Case(
-                models.When(forward, then=models.F('value')),
-                models.When(reverse, then=models.Value(get_one()) / models.F('value')),
-            )
-        ).get(forward | reverse, backend=backend).rate
-    except Rate.DoesNotExist:
+    rates = Rate.objects.filter(currency__in=(source, target), backend=backend)
+    if not rates:
         raise MissingRate('Rate %s -> %s does not exist' % (source, target))
+    # Direct rate
+    if len(rates) == 1:
+        rate = rates[0]
+        if text_type(rate.currency) == text_type(target):
+            return rate.value
+        return 1 / rate.value
+    # Indirect rate
```

Comment:
So an "indirect rate" means to convert from rate A to rate C by first converting to rate A to B, then B to C? Maybe a few comments in this code to clarify what's going on :)

Reply:
For example, we want to convert NOK to SEK and we have 2 rates - NOK -> USD and SEK -> USD. 
E.g. 1 USD = 7.84 NOK and 1 USD = 8.37 SEK -> 7.84 NOK = 8.37 SEK -> 1 NOK = 8.37 / 7.84 SEK

For the case when we need to convert NOK to USD, we just don't need the second item. Seems like "direct" and "indirect" terms are confusing, I'll try to rephrase it :)